### PR TITLE
GODRIVER-2811 Use UTC when marshalling primitive.DateTime as JSON.

### PR DESCRIPTION
**Pull request by dnikulin**
[GODRIVER-2811](https://jira.mongodb.org/browse/GODRIVER-2811)

## Summary
Make the JSON format of `primitive.DateTime` use UTC, consistent with the relaxed extJSON format.

## Background & Motivation
Marshalling in relaxed extJSON already uses UTC, which makes sense since the type has no timezone information.
Using local time in JSON does not appear to have been intentional, as there is no comment or test for this behavior.
Using UTC allows the JSON to be deterministic with respect to the BSON, rather than depend on the environment.

### CODE REVIEW

**Review by matthewdale — APPROVED** [review]
@dnikulin thanks for the PR! Looks good 👍 

I've created a GODRIVER Jira ticket to track this behavior change and added it to the title/description.

**Comment by dnikulin** [comment]
Thank you @matthewdale !
By the way, I don't have write access. Could you please merge when convenient?

**Comment by matthewdale** [comment]
@dnikulin I've added a 2nd reviewer. As soon as they approve it, I'll merge the PR.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**
